pkg/node/node.go

```diff
@@ -54,10 +54,27 @@ type NodeConfig struct {
 	FailureInjectionConfig models.FailureInjectionConfig
 }
 
+// Validate Config
 func (c *NodeConfig) Validate() error {
 	// TODO: add more validations
 	var mErr error
 	mErr = errors.Join(mErr, validate.NotBlank(c.NodeID, "node id is required"))
+
+	//// Check that Auth and BuiltInAuth are not both defined
+	//authDefined := len(c.BacalhauConfig.API.Auth.Methods) > 0 || c.BacalhauConfig.API.Auth.AccessPolicyPath != ""
+	//
+	//builtInAuthDefined := false
+	//if c.BacalhauConfig.API.BuiltInAuth.BasicAuth.Admins.SrcFilePath != "" ||
+	//	c.BacalhauConfig.API.BuiltInAuth.BasicAuth.Users.SrcFilePath != "" ||
+	//	len(c.BacalhauConfig.API.BuiltInAuth.BasicAuth.Admins.UserConfig) > 0 ||
+	//	len(c.BacalhauConfig.API.BuiltInAuth.BasicAuth.Users.UserConfig) > 0 {
+	//	builtInAuthDefined = true
+	//}
+	//
+	//if authDefined && builtInAuthDefined {
+	//	mErr = errors.Join(mErr, fmt.Errorf("both Auth and BuiltInAuth are defined in the configuration; only one can be used at a time"))
+	//}
+
```

Comment:
do we need this?

Reply:
This was a TODO item for me to add some validation, which was added now.